packages/insomnia-components/src/button/async-button.tsx

```diff
@@ -14,7 +14,7 @@ function isPromise(obj: unknown) {
 }
 
 export interface AsyncButtonProps<T> extends ButtonProps {
-  onClick: (e: React.MouseEvent<HTMLButtonElement>) => Promise<T>;
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => Promise<T> | undefined;
```

Comment:
```suggestion
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => Promise<T>;
```

Should this be this?  It doesn't make a difference now, but it will when we get to TS 4.4.

Reply:
No no these are slightly different (and very hard to see). This is defining the `onClick` handler as required, but returning `Promise<T> | undefined`

Reply:
(At least that is my intent, but if the syntax is incorrect I'll fix)

Reply:
ahhh I was reading it too fast.  thanks.  cool.